src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_FACTOR = 0.75f;
+    private int currentCapacity;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+        this.currentCapacity = table.length;
+        this.threshold = (int) (currentCapacity * DEFAULT_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = getIndex(key);
+        if (size >= threshold) {
+            resize();
+        }
+        putValue(index, newNode, table);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (key == null && currentNode.key == null
+                    || key != null && key.equals(currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] newTable = new Node[table.length << 1];
+        currentCapacity = newTable.length;
+        threshold = (int) (currentCapacity * DEFAULT_FACTOR);
+        size = 0;
+        table = transferData(table, newTable);
+    }
+
+    private Node<K, V>[] transferData(Node<K, V>[] oldTab, Node<K, V>[] newTab) {
```

Comment:
The 'transferData' method should have a private access modifier since it's only used within the class itself.